### ISSUE TITLE
V2 Error/not working

### ISSUE THREAD

**Issue by Yokopops** [automatic-ripping-machine/automatic-ripping-machine#188]
Finally got around to installing V2, it worked yesterday. However, it would only work on 1 episode from a series and now it wont work at all and give out this error:

[2018-06-20 22:52:06] INFO ARM: Starting ARM processing at 2018-06-20 22:52:06.055477
[2018-06-20 22:52:06] INFO ARM: ARM version: 2.0.0-beta
[2018-06-20 22:52:06] INFO ARM: Python version: 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609]
[2018-06-20 22:52:06] INFO ARM: Looking for log files older than 1 days old.
[2018-06-20 22:52:06] INFO ARM: Starting Disc identification
[2018-06-20 22:52:06] INFO ARM: Mounting disc to: /mnt/dev/sr0
[2018-06-20 22:52:06] INFO ARM: Disc identified as video
[2018-06-20 22:52:06] INFO ARM: Getting movie title...
[2018-06-20 22:52:06] INFO ARM: DVD CRC64 hash is: fdc776a783143668
[2018-06-20 22:52:07] ERROR ARM: A fatal error has occured and ARM is exiting.  See traceback below for details.
Traceback (most recent call last):
  File "/opt/arm/arm/getmovietitle.py", line 88, in main
    disc_title, disc_year = getdvdtitle(disc)
  File "/opt/arm/arm/getmovietitle.py", line 40, in getdvdtitle
    dvd_title = doc['METADATA']['MDR-DVD']['dvdTitle']
KeyError: 'dvdTitle'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/arm/arm/main.py", line 225, in <module>
    main(logfile, disc)
  File "/opt/arm/arm/main.py", line 85, in main
    identify.identify(disc, logfile)
  File "/opt/arm/arm/identify.py", line 51, in identify
    disc.videotitle, disc.videoyear = getmovietitle.main(disc)
  File "/opt/arm/arm/getmovietitle.py", line 92, in main
    disc_title, disc_year = getbluraytitle(disc)
  File "/opt/arm/arm/getmovietitle.py", line 52, in getbluraytitle
    with open(disc.mountpoint + '/BDMV/META/DL/bdmt_eng.xml', "rb") as xml_file:
FileNotFoundError: [Errno 2] No such file or directory: '/mnt/dev/sr0/BDMV/META/DL/bdmt_eng.xml'

**Comment by muckngrind4** [comment]
It looks like the DVD isn't in Microsoft's Windows Media database. Is it a factory DVD or a copy?  I need to add logic to fail gracefully in these situations. I'll flag this as a bug.

**Comment by Yokopops** [comment]
Hi, It was a factory DVD. I have tried the first DVD in my Buffy box set and it ripped it the first time then failed, the second DVD was a Family Guy season 5 an that never worked, both of these previously ripped fine on V1 as I used these DVDs as benchmarks if you will. I am now attempting to rip a couple of films that I have previously ripped on V1 so it rules out box sets (Going to try a couple of others that worked on V1). The other thing is when it rips DVDs it appears to not name the files only the folders? The file name is just Title00, 01,02 and so on is this expected behaviour? (Not sure if it does the same on V2 yet)

Kind regards.

P.s How do I launch the webserver, Just run the startweb.py, and is it safe to run at startup?

**Comment by Yokopops** [comment]
Update: Tried Star Trek Into Darkness and it worked, however buffy still is a no go and seems to lock the dvd drive closed, Family guy season 1 disc 1 is been named to blue harvest (will try the second dvd after). Is there a way to make it still rip the dvd if it cannot find it via Microsoft's database? I believe that is how V1 was working for me as it never named any of the files.

**Comment by muckngrind4** [comment]
@Yokopops I've pushed a fix to the v2_fixes branch.  If you don't mind, would you checkout that branch and try your dvds?

**Comment by Yokopops** [comment]
@muckngrind4 I have been out for a couple of days so will attempt to update now and get back to you. I just swap the branch from V2 to the V2 fixes branch like swapping from V1 to V2 I take it?

**Comment by muckngrind4** [comment]
I just merged it into V2_master so you should be able to just update (pull) the current branch.

**Comment by Yokopops** [comment]
I already updated to the fixes branch, sorry. I have attempted my Buffy disc 1 and waiting to see if it encodes all episodes or only the first episode again, will attempt family guy after htop reports no activity. My webserver also appears to not display correctly as it displays too many numbers on space left like 852.63243241GB and does not display the queue, will check it on this branch and see if its the same, will keep you posted.

Kind regards.

**Comment by muckngrind4** [comment]
No worries. Both branches are the same right now. :)  Let me know how it goes.

The webserver stuff is really experimental right now.  It only works if you're logging everything to empty.log, which isn't the default.  I'm targeting 2.1 for something a little more robust.  I may have some very preliminary stuff up in the next week or two on that branch.

**Comment by Yokopops** [comment]
Update: It ripped the whole DVD however it only encoded 1 episode and named it "Buffy the Vampire Slayer - Season 1" (The same as the folder name) however in the RAW folder I have title00.mkv to title03.mkv

**Comment by Yokopops** [comment]
@muckngrind4 Family guy disc 1 is doing the same, naming to blue harvest and only encoding first episode and naming the file the same as the folder (Same thing as Buffy)

**Comment by muckngrind4** [comment]
Can you turn on debugging (in arm.yaml file), run it and post the full config?

**Comment by Yokopops** [comment]
Post the full .yaml file I take it? Also what would you want me to run, the Buffy DVD? I then guess you would like the Buffy log file, correct?

**Comment by muckngrind4** [comment]
I don't really need the yaml file. Just the log file after you've run a problem DVD. Buffy would be a good one.   But do make sure to set the log level to DEBUG in the yaml file before running it.

**Comment by Yokopops** [comment]
Its currently processing want me to wait until its no longer running in htop? Or is the info you need early on in the file?

**Comment by muckngrind4** [comment]
The debug setting needs to be set prior to starting the run.

**Comment by Yokopops** [comment]
Already done that 👍

**Comment by Yokopops** [comment]
Where do I post the log as its 52002 lines long?

**Comment by muckngrind4** [comment]
If you drag and drop it on the comment box it should attach to you comment.

**Comment by Yokopops** [comment]
Sorry for been dumb. Should have seen that, lmao
[BUFFY_DISC1.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/2146414/BUFFY_DISC1.log)

**Comment by muckngrind4** [comment]
Ok. I see a couple issues--one is a bad response from the Windows Media service which causes the title to be identified as a movie instead of a series. Its a minor issue that is easily worked around 
however, the other one is a bug. I'll work on that this weekend and let you know when I've got a fix up. Are you still pointed at v2_fixes?

**Comment by Yokopops** [comment]
@muckngrind4 Yeah I am still on V2_fixes. glad to be of use on this project i'm really impressed with how it functions. If I can be of any other use don't hesitate to tell me.

**Comment by muckngrind4** [comment]
@Yokopops I've updated v2_fixes if you want to give it a try. Let me know your results.

**Comment by muckngrind4** [comment]
Fixed in #192

**Comment by Yokopops** [comment]
@muckngrind4 I will run the tests now as I have only just gotten in will post back in a while.

**Comment by Yokopops** [comment]
@muckngrind4 Season 1 worked fine, however, it is not even generating a log or anything for season 2 disc 1. I am currently attempting to correct this my end.

**Comment by Yokopops** [comment]
@muckngrind4 Ok, so I have gotten it to rip disc 1, however, it will not start to rip disc 2 while processing disc one. also i'm not getting a log file. It is as if it does not detect the disc has been inserted.

**Comment by Yokopops** [comment]
Tried another DVD and it is refusing to acknowledge a new DVD to rip while it is processing.

**Comment by Yokopops** [comment]
@muckngrind4 Will upload the log later. It is naming a Buffy disc Pacific Rim now haha

**Comment by Yokopops** [comment]
[BUFFY_SEASON2_DISC2.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/2168088/BUFFY_SEASON2_DISC2.log)

Is there enough info in here?

**Comment by muckngrind4** [comment]
That's a weird one.  I checked the Microsoft meta service and it's definitely coming back as Pacific Rim.  There isn't really anything we can do about that.  It's how MS has it in their db.

There is a setting called VIDEOTYPE.  Set it to "series" and ARM will at least handle it like a tv series and not a movie.  

As for the snippet of log you posted, that's the output from HandBrake and it's very verbose.  Most of it is way above my understanding, and it will output a lot of stuff that looks bad but is really just normal.  From my limited understanding, I don't see any errors above, so I don't think that is where your problem is.  

If you don't mind attaching the whole log I'll parse it out.  I'm very curious to see if my fix is still broken.

**Comment by Yokopops** [comment]
[BUFFY_SEASON2_DISC4.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/2173208/BUFFY_SEASON2_DISC4.log)

Disc 4 log (Cannot rip other discs while processing, also locks dvd drive shut, have to send eject command to open.)

**Comment by Yokopops** [comment]
Just noticed it is not removing the RAW folders either.

**Comment by Yokopops** [comment]
@muckngrind4 Yeah it's all DVDs I put in The only thing I have yet to try is a reinstall of the entire server and I would like to avoid that if possible, if that is the last resort I will bite the bullet. I cannot currently insert a new DVD until the previous has finished processing and there is nothing running in the process list

**Comment by muckngrind4** [comment]
I think the I/O errors and buffer errors in your log snippet above are likely indicators of the problem.  A quick search seemed to indicate disc or drive failures.  If it's all DVD's then that would further indicate a drive issue.  But, I'm no expert in hardware stuff by any means...

FYI, thanks for the heads up on the raw folders.  It's fixed now.

**Comment by Yokopops** [comment]
I will attempt a reinstall then and go from there I guess. I know its not a drive issue as it was only after I swapped branch, as the issue I had before was the naming issue. Is there a way to totally remove ARM and start fresh or have I got to totally nuke my server?

**Comment by Yokopops** [comment]
@muckngrind4 should I stay on v2_fixes or go to v2 master?

**Comment by Yokopops** [comment]
@muckngrind4 Just got this error at the end of one of my logs:
 [2018-07-14 17:32:52] ERROR ARM: main.<module> A fatal error has occured and ARM is exiting.  See traceback below for details.
Traceback (most recent call last):
  File "/opt/arm/arm/main.py", line 232, in <module>
    main(logfile, disc)
  File "/opt/arm/arm/main.py", line 102, in main
    if cfg['HASHEDKEYS']:
KeyError: 'HASHEDKEYS'

Think I just need to reinstall...

**Comment by muckngrind4** [comment]
@Yokopops That's an easy one.  The newest commit requires an updated entry in your config.  Look at the arm.yaml.sample [here](https://github.com/automatic-ripping-machine/automatic-ripping-machine/blob/v2_master/docs/arm.yaml.sample) and copy the HASHEDKEYS line into your existing arm.yaml file...

**Comment by muckngrind4** [comment]
PS. I'm going to close this issue since it's got some bugs filed against it that are fixed.  Feel free to start a new issue if you find other issues.

**Comment by Yokopops** [comment]
@muckngrind4 Thanks for that, will try now. Sorry for been a total pain...

**Comment by muckngrind4** [comment]
@yokopops No problem. I appreciate the feedback!
